File: Backend API/Locafest/com.locafest.Domain/LocacaoContext/Entities/Agendamento.cs
using com.locafest.Domain.UsuarioContext.Entities;
using System;
using System.Collections.Generic;
using System.Text;
using System.Text.Json.Serialization;

namespace com.locafest.Domain.LocacaoContext.Entities
{
    public class Agendamento : BaseEntity
    {
        public DateTime DataRetirada { get; set; }
        public DateTime DataDevolucao { get; set; }
        public string Local { get; set; }
        public decimal ValorHora { get; set; }
        public decimal QuantidadeHoras { get; set; }
        public decimal ValorFinal { get; set; }
        public Guid VeiculoId { get; set; }
        public Guid ClienteId { get; set; }
        [JsonIgnore]
        public Veiculo Veiculo { get; set; }
        [JsonIgnore]
        public Cliente Cliente { get; set; }
    }
}

File: Mobile FrontEnd/locafest/lib/app/shared/entities/agendamento.dart
class Agendamento {
  String id;
  String dataRetirada;
  String dataDevolucao;
  String local;
  int valorHora;
  int quantidadeHoras;
  int valorFinal;
  String veiculoId;
  String clienteId;

  Agendamento(
      {this.id,
      this.dataRetirada,
      this.dataDevolucao,
      this.local,
      this.valorHora,
      this.quantidadeHoras,
      this.valorFinal,
      this.veiculoId,
      this.clienteId});

  Agendamento.fromJson(Map<String, dynamic> json) {
    id = json['id'];
    dataRetirada = json['dataRetirada'];
    dataDevolucao = json['dataDevolucao'];
    local = json['local'];
    valorHora = json['valorHora'];
    quantidadeHoras = json['quantidadeHoras'];
    valorFinal = json['valorFinal'];
    veiculoId = json['veiculoId'];
    clienteId = json['clienteId'];
  }

  Map<String, dynamic> toJson() {
    final Map<String, dynamic> data = new Map<String, dynamic>();
    data['id'] = this.id;
    data['dataRetirada'] = this.dataRetirada;
    data['dataDevolucao'] = this.dataDevolucao;
    data['local'] = this.local;
    data['valorHora'] = this.valorHora;
    data['quantidadeHoras'] = this.quantidadeHoras;
    data['valorFinal'] = this.valorFinal;
    data['veiculoId'] = this.veiculoId;
    data['clienteId'] = this.clienteId;
    return data;
  }
}